topdown/http.go

```diff
@@ -895,10 +895,15 @@ type interQueryCacheData struct {
 	Headers    http.Header
 }
 
-func newInterQueryCacheData(resp *http.Response, respBody []byte) (*interQueryCacheData, error) {
-	_, err := parseResponseHeaders(resp.Header)
-	if err != nil {
-		return nil, err
+func newInterQueryCacheData(resp *http.Response, respBody []byte, force bool) (*interQueryCacheData, error) {
+	if force {
+		now := time.Now().UTC()
+		resp.Header["Date"] = []string{now.Format(http.TimeFormat)}
```

Comment:
I think I understand how this solves the issue, but wouldn't it also make sense to be more explicit about it? I'd imagine a flow like this:

> To get the TTL, we'll either add some duration to the Date header, or take the duration from now.

_Setting_ the Date header for the force-cache cache seems a bit indirect 🤔  What do you think?

Reply:
The date header of the response is what later decides whether the cache entry is "fresh" or not. We cannot add some duration to the date header provided by the remote server as:

1. It might not be there at all (which is the cause of the issue).
2. It might be there, but not provide an accurate date. If we trust that, we're effectively not forcing anything, but being forced ;)

Reply:
So maybe we shouldn't let the header matter at all in the TTL decision...? 🤔

Reply:
So rather than storing the Date header, we'd add a "createdAt" field to the struct storing the `time.Time` directly? In the case of `force_cache: true` this would always be created from the current time, and if else from the Date header? Should we still include the Date header in the cached response data then? 🤔 It would be rather nice to rewite the logic and abstract away the HTTP bits better, but that's something I'd rather do _after_ the test suite has been made more extensive 😅

Reply:
Fair enough. :)